compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -1144,7 +1148,7 @@ object Semantic:
 
   extension (arg: ArgInfo)
     def promote: Contextual[Unit] = withTrace(arg.trace) {
-      arg.value.promote("Cannot prove the method argument is hot. Only hot values are safe to leak.\nFound = " + arg.value.show + ".")
+      arg.value.promote("Could not verify that the method argument is transitively initialized (Hot). It was found to be " + arg.value.show + ". Only transitively initialized arguments may be passed to methods outside the analyzed initialization region.")
```

Comment:
This is inaccurate, I'd suggest remove `outside the analyzed initialization region` and add `(except constructors)`.

Reply:
We allow passing non-hot arguments to constructors, but not methods.